src/components/NewUrl.vue

```diff
@@ -31,27 +34,65 @@
 
 export default {
   name: 'NewUrl',
-  props: ['editUrl'],
-  data: function () {
-    console.log(`data:${this.editUrl}`);
+
+  props: {
+    url: {
+      type: Object,
+      required: true,
+    },
+  },
+
+  data() {
     return {
-      url: this.editUrl
+      updatedUrl: {...this.url},
     }
   },
   methods: {
     addNewUrl: function () {
-      if (!this.url.url || !this.url.short_url) {
+      if (!this.updatedUrl.url || !this.updatedUrl.short_url) {
         alert('url and short_url can not be empty');
         return;
       }
+
+      const apiUrl = process.env.VUE_APP_API_BASE_URL;
```

Comment:
Why don't declare API_URL once before export, I mean exactly the same string but once at the start